### PR TITLE
Add backwards compatibility support to get_highest_snapshot_slot()

### PR DESCRIPTION
**Pull request by brooksprumo**
As per this comment from @mvines (https://github.com/solana-labs/solana/pull/19587#issuecomment-912036981), adding backwards compatibility for the `get_highest_snapshot_slot()` RPC call.

### CODE REVIEW

**Comment by mvines** [comment]
👮🏼  Why you asking me to review a Draft PR?

File: client/src/rpc_client.rs [hunk 1]
```diff
@@ -1041,7 +1041,16 @@ impl RpcClient {
     /// # Ok::<(), ClientError>(())
     /// ```
     pub fn get_highest_snapshot_slot(&self) -> ClientResult<RpcSnapshotSlotInfo> {
-        self.send(RpcRequest::GetHighestSnapshotSlot, Value::Null)
+        if self.get_node_version()? < semver::Version::new(1, 8, 0) {
+            #[allow(deprecated)]
+            let full_snapshot_slot = self.get_snapshot_slot()?;
+            Ok(RpcSnapshotSlotInfo {
+                full: full_snapshot_slot,
```
Comment (mvines) [inline client/src/rpc_client.rs:1048]: nit:
```suggestion
            Ok(RpcSnapshotSlotInfo {
                full: self.get_snapshot_slot()?,
```

**Review by mvines — APPROVED** [review]

File: client/src/rpc_client.rs [hunk 2]
```diff
@@ -1041,7 +1041,15 @@ impl RpcClient {
     /// # Ok::<(), ClientError>(())
     /// ```
     pub fn get_highest_snapshot_slot(&self) -> ClientResult<RpcSnapshotSlotInfo> {
-        self.send(RpcRequest::GetHighestSnapshotSlot, Value::Null)
+        if self.get_node_version()? < semver::Version::new(1, 8, 0) {
+            #[allow(deprecated)]
+            Ok(RpcSnapshotSlotInfo {
+                full: self.get_snapshot_slot()?,
+                incremental: None,
+            })
```
Comment (mvines) [inline client/src/rpc_client.rs:1049]: oh wait, even better!
```rust
self.get_snapshot_slot().map(|full| RpcSnapshotSlotInfo { full, incremental: None })
```

Comment (brooksprumo) [reply client/src/rpc_client.rs]: Love it. Done!

**Review by mvines — APPROVED** [review]

**Pull request merged**